src/libraries/System.Private.CoreLib/src/System/Collections/Generic/ArraySortHelper.cs

```diff
@@ -321,12 +325,30 @@ internal partial class GenericArraySortHelper<T>
 
         #region IArraySortHelper<T> Members
 
-        public void Sort(Span<T> keys, IComparer<T>? comparer)
+        public unsafe void Sort(Span<T> keys, IComparer<T>? comparer)
         {
             try
             {
                 if (comparer == null || comparer == Comparer<T>.Default)
                 {
+
+                    if (keys.Length == 0) {
+                        return;
+                    }
+
+                    if (Avx2.IsSupported)
+                    {
+                        if (typeof(T) == typeof(int))
+                        {
+                            fixed (int* startPtr = &Unsafe.As<T, int>(ref keys[0]))
+                            {
+                                var sorter = new VectorizedSort.VectorizedUnstableSortInt32(startPtr, keys.Length);
```

Comment:
Reading through the paper and blog, it looks like the algorithms for different sizes are basically the same, just with different lookup tables and increment sizes. Is that correct?

Reply:
I assume that by different sizes you mean different types?

If so, then:
* unsigned sorting requires two lookup-tables (at least according to my plan), per size, but one of those tables is the one used by signed sorting.
  * I can get into this when we want to dive into my "plans" for unsigned sorting.
* `float` would share the table with `int`
* and so one per type/size in bits, so:
  * Two more tables to support `long`/`ulong`
  * an so on per 16, 8 bits

Does that answer your question?

Reply:
The question was more around if the primary difference between `int` and `short` was just around the lookup table and not necessarily the algorithm used for sorting (other than the number of elements processes "per iteration")?

Trying to understand if supporting the other types would result in a code explosion or if most of the algorithm could be shared, just passing in different increment sizes and lookup tables (and calling a different instruction for the permutations).